perma_web/api/views.py

```diff
@@ -195,8 +195,14 @@ def post(self, request, format=None):
         data = request.data.copy()
         if request.parent:
             data.setdefault('parent', request.parent.pk)
-
-        return self.simple_create(data, {'created_by': request.user})
+        with transaction.atomic():
+            if data.get('parent'):
+                # Lock the parent to prevent anyone from deleting it while this operation is validated and saved.
+                parent = Folder.objects.select_for_update().get(pk=data['parent'])
+                # We don't want to insert any new folders while a tree's folders are being moved around.
+                # Since moves lock the whole subtree, we can take out the same lock, to ensure no move is underway.
```

Comment:
I'm not sure what "take out the same lock" means in this context.

Reply:
No, that makes sense. Thanks!

Reply:
if you had a `parent.lock_tree()` method that did `Folder.objects.select_for_update().get(pk=self.tree_root_id)`, you could use it in a few places and maybe some comments would seem superfluous. nbd!

Reply:
Ah, that's a good idea. Might refactor in a future PR!

Thanks for taking a glance.